dotnet/src/Connectors/Connectors.AssemblyAI/AudioToText/AssemblyAIAudioToTextService.cs

```diff
@@ -0,0 +1,292 @@
+        var json = await JsonDocument.ParseAsync(jsonStream, cancellationToken: ct).ConfigureAwait(false);
+        if (json.RootElement.TryGetProperty("error", out var property))
+        {
+            throw new AssemblyAIApiException($"Failed to create transcript. Reason: {property.GetString()!}");
+        }
+
+        return json.RootElement.GetProperty("id").GetString()!;
+    }
+
+    private async Task<JsonDocument> WaitForTranscriptToProcessAsync(string transcriptId, CancellationToken ct)
+    {
+        var url = $"https://api.assemblyai.com/v2/transcript/{transcriptId}";
+
+        while (!ct.IsCancellationRequested)
+        {
+            using var request = new HttpRequestMessage(HttpMethod.Get, url);
+            request.Headers.Authorization = new AuthenticationHeaderValue(this._apiKey);
+            using var response = await this._httpClient.SendAsync(request, ct).ConfigureAwait(false);
+            await ThrowIfNotSuccessStatusCodeAsync("Error waiting for transcript.", response, ct)
+                .ConfigureAwait(false);
+            var jsonStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+            var json = await JsonDocument.ParseAsync(jsonStream, cancellationToken: ct).ConfigureAwait(false);
+
+            var status = json.RootElement.GetProperty("status").GetString()!;
+            switch (status)
+            {
+                case "processing":
+                case "queued":
+                    await Task.Delay(TimeSpan.FromSeconds(1), ct).ConfigureAwait(false);
+                    break;
+                case "completed":
+                    return json;
+                case "error":
+                    var errorString = json.RootElement.GetProperty("error").GetString()!;
+                    throw new AssemblyAIApiException($"Failed to create transcript. Reason: {errorString}");
+                default:
+                    throw new AssemblyAIApiException("Unexpected transcript status. This code shouldn't be reachable.");
+            }
+        }
+
+        ct.ThrowIfCancellationRequested();
+        throw new AssemblyAIApiException("This code is unreachable.");
+    }
+
+    private static async Task ThrowIfNotSuccessStatusCodeAsync(
+        string errorMessagePrefix,
+        HttpResponseMessage response,
+        CancellationToken ct
+    )
+    {
+        if (response.IsSuccessStatusCode)
+        {
+            return;
+        }
+
+        if (response.Content.Headers.ContentType.MediaType == "application/json")
+        {
+            var jsonStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+            var json = await JsonDocument.ParseAsync(jsonStream, cancellationToken: ct).ConfigureAwait(false);
+            if (json.RootElement.TryGetProperty("error", out var property))
+            {
+                throw new AssemblyAIApiException($"{errorMessagePrefix} Reason: {property.GetString()!}");
```

Comment:
If it is sufficient throws any common .net exception if not use KernelException not custom.